swim/memberlist_test.go

```diff
@@ -338,6 +342,114 @@ func (s *MemberlistTestSuite) TestApplyUnknownTombstone() {
 	s.Assert().Len(applied, 0, "expected that the declaration of a tombstone for an unknown member is not applied")
 }
 
+func (s *MemberlistTestSuite) TestSetLocalLabel() {
+	// make sure there are no changes recorded before the test
+	s.node.disseminator.ClearChanges()
+
+	s.m.SetLocalLabel("hello", "world")
+	value, has := s.m.GetLocalLabel("hello")
+
+	s.Assert().True(has, "expected to have the local label hello after is has been set")
+	s.Assert().Equal("world", value, "expected a value of world for the local label")
+	s.Assert().Equal(1, s.node.disseminator.ChangesCount(), "expected to have 1 change recorded in the disseminator")
+
+	// testing to overwrite the value
+	s.m.SetLocalLabel("hello", "baz")
+	value, has = s.m.GetLocalLabel("hello")
+
+	s.Assert().True(has, "expected to have the local label hello after is has been overwritten")
+	s.Assert().Equal("baz", value, "expected a value of baz for the local label")
+	s.Assert().Equal(1, s.node.disseminator.ChangesCount(), "expected to have 1 change recorded in the disseminator after the labels are overwritten")
```

Comment:
I think I see a race here: from the time label is set until the `disseminator.ChangesCount()` is called. What happens if during the time the dissemination period kicks in and the change is disseminated? Or we are disabling pings somewhere, and I can't see it (likely)?